python/pyarrow/types.pxi

```diff
@@ -529,6 +529,49 @@ cdef class Schema:
     def get_field_index(self, name):
         return self.schema.GetFieldIndex(tobytes(name))
 
+    def add_field(self, int i, Field field):
```

Comment:
This feels like just a wrapping of the C++ API. Is there an API that we can provide that'll be more familiar to Python users? How about `insert`/`append`/`remove` similar to `list`s?

Reply:
Either way I think these methods need to return new objects, so Schema is not mutable like list, though

Reply:
Sure. I don't think the semantics need to be identical, but I think the spelling can be improved.

Reply:
As this is most similar to a list, I would use the function names as defined there but would do some adjustments, e.g.:

 * Always return a new object
 * Use indices instead of actual values, i.e. "`remove(x)` would be `remove(i)`".